### ISSUE TITLE
v1.23 does not work with SonarQube 25.3.0.104237-community

### ISSUE THREAD

**Issue by gschafra** [mc1arke/sonarqube-community-branch-plugin#1049]
**Describe the bug**
Plugin v1.23 does not work with SonarQube 25.3.0.104237-community (see also [comment](https://github.com/mc1arke/sonarqube-community-branch-plugin/issues/1033#issuecomment-2701865187) on #1033) . No error is shown in the logs.
May be caused by massive code (and API?) changes in the SonarQube v25.2 -> v25.3 (see https://github.com/SonarSource/sonarqube/compare/25.2.0.102705...25.3.0.104237) and/or Plugin-API v11.1 -> v11.2 (see https://github.com/SonarSource/sonar-plugin-api/compare/11.1.0.2693...11.2.0.2797)  update

**To Reproduce**
Just use image (tag 25.3) provided by https://github.com/adrianmusante/docker-sonarqube/tree/main (https://github.com/adrianmusante/docker-sonarqube/tree/main). Plugin seems to be loaded correctly, see following excerpt from sonar.log:
```log
...
sonarqube-1  | 2025.03.10 08:14:32 INFO  web[][o.s.c.e.CoreExtensionsLoader] Loaded core extensions: Community Branch Plugin
sonarqube-1  | 2025.03.10 08:14:32 INFO  web[][o.s.s.p.LogServerVersion] SonarQube Server / 25.3.0.104237 / 7cc57fb683cea7b8fbdfc3965ab93ebfd2916c60
...
sonarqube-1  | 2025.03.10 08:14:33 INFO  web[][o.s.s.p.ServerPluginManager] Deploy Community Branch Plugin / 1.23.0 / null
...
sonarqube-1  | 2025.03.10 08:14:38 INFO  web[][o.s.s.p.p.PlatformLevelStartup] Running Community Edition
sonarqube-1  | 2025.03.10 08:14:38 INFO  web[][o.s.s.p.Platform] Web Server is operational
sonarqube-1  | 2025.03.10 08:14:38 INFO  app[][o.s.a.SchedulerImpl] Process[web] is up
sonarqube-1  | 2025.03.10 08:14:38 INFO  app[][o.s.a.ProcessLauncherImpl] Launch process[COMPUTE_ENGINE] from [/opt/bitnami/sonarqube]: /opt/bitnami/java/bin/java -Djava.awt.headless=true -Dfile.encoding=UTF-8 -Djava.io.tmpdir=/opt/bitnam
i/sonarqube/temp -XX:-OmitStackTraceInFastThrow --add-opens=java.base/java.util=ALL-UNNAMED --add-exports=java.base/jdk.internal.ref=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.nio=ALL-UNNAMED --add-
opens=java.base/sun.nio.ch=ALL-UNNAMED --add-opens=java.management/sun.management=ALL-UNNAMED --add-opens=jdk.management/com.sun.management.internal=ALL-UNNAMED -Xmx512m -Xms128m -XX:+HeapDumpOnOutOfMemoryError -javaagent:/bitnami/sonarqu
be/extensions/plugins/sonarqube-community-branch-plugin.jar=ce -Dhttp.nonProxyHosts=localhost|127.*|[::1] -cp ./lib/sonar-application-25.3.0.104237.jar:/opt/bitnami/sonarqube/lib/jdbc/postgresql/postgresql-42.7.5.jar org.sonar.ce.app.CeSe
rver /opt/bitnami/sonarqube/temp/sq-process17601094284425973468properties
sonarqube-1  | 08:14:38.720 [main] INFO com.github.mc1arke.sonarqube.plugin.CommunityBranchAgent -- Loading agent
...
sonarqube-1  | 2025.03.10 08:14:38 INFO  ce[][o.s.c.e.CoreExtensionsLoader] Loaded core extensions: Community Branch Plugin
...

```

- Analyzed project's branch selector (dropdown)  is no longer functional (see screenshot)
- Missing "Branches & Pull Requests" Project Settings Entry. Navigating to the corresponding settings page manually causes "The page you were looking for does not exist." to be shown (without any corresponding entry in the logs)

**Expected behavior**

- Analyzed branches selectable
- Branch analysis possible

**Screenshots**
![Image](https://github.com/user-attachments/assets/16e99920-11ec-4db6-b099-7bf21d9a835f)

**Software Versions**
 - SonarQube Version: 25.3.0.104237-community
 - Plugin Version: 1.23

**Comment by mc1arke** [comment]
I think this is likely to be https://github.com/SonarSource/sonarqube-webapp/commit/d5214e1472063e4985ffc70eb59fda0fa30dbbe9 which has removed parts of the branch UI from the community edition front-end. Whilst plugins can contribute pages to the UI, I don't know if it will be possible to inject components back into the core framework. Interestingly it is possibly to still load issues on branches and pull requests by adding a `pullRequest={id}` or `branch={id}` parameter to the URL, so the data is still being read from the APIs properly - but the overview don't load from a quick test so I presume there have been other front-end changes to limit the data that's being retrieved or rendered for some operations.

I'll have a go at seeing what it's possible to hack together, but don't hold out to much hope of this working or being a long-term viable solution. Which me luck!

**Comment by killallxxx** [comment]
Добрый день! SonarQube Community Build v25.3.0.104237
Добавил плагин [1.23.0](https://github.com/mc1arke/sonarqube-community-branch-plugin/releases/tag/1.23.0) При запуске сервера получаю следующаю ошибку и остановку сервера. Кто сталкивался, что подскажите? 
Может по советуете рабочую версию сервера и совместимую с ней версию плагина?
Starting SonarQube...
2025.03.16 16:27:38 INFO  app[][o.s.a.AppFileSystem] Cleaning or creating temp directory c:\tools\sonarqube\temp
2025.03.16 16:27:38 INFO  app[][o.s.a.es.EsSettings] Elasticsearch listening on [HTTP: 127.0.0.1:9001, TCP: 127.0.0.1:{}]
2025.03.16 16:27:38 INFO  app[][o.s.a.ProcessLauncherImpl] Launch process[ELASTICSEARCH] from [C:\tools\sonarqube\elasticsearch]: C:\Program Files\Axiom\AxiomJDK-Pro-17-Full\bin\java -Xms4m -Xmx64m -XX:+UseSerialGC -Dcli.name=server -Dcli.script=./bin/elasticsearch -Dcli.libs=lib/tools/server-cli -Des.path.home=C:\tools\sonarqube\elasticsearch -Des.path.conf=c:\tools\sonarqube\temp\conf\es -Des.distribution.type=tar -cp C:\tools\sonarqube\elasticsearch\lib\*;C:\tools\sonarqube\elasticsearch\lib\cli-launcher\* org.elasticsearch.launcher.CliToolLauncher
2025.03.16 16:27:38 INFO  app[][o.s.a.SchedulerImpl] Waiting for Elasticsearch to be up and running
Standard Commons Logging discovery in action with spring-jcl: please remove commons-logging.jar from classpath in order to avoid potential conflicts
2025.03.16 16:28:06 INFO  app[][o.s.a.SchedulerImpl] Process[es] is up
2025.03.16 16:28:06 INFO  app[][o.s.a.ProcessLauncherImpl] Launch process[WEB_SERVER] from [C:\tools\sonarqube]: C:\Program Files\Axiom\AxiomJDK-Pro-17-Full\bin\java -Djava.awt.headless=true -Dfile.encoding=UTF-8 -Djava.io.tmpdir=c:\tools\sonarqube\temp -XX:-OmitStackTraceInFastThrow --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED --add-exports=java.base/jdk.internal.ref=ALL-UNNAMED --add-opens=java.base/java.nio=ALL-UNNAMED --add-opens=java.base/sun.nio.ch=ALL-UNNAMED --add-opens=java.management/sun.management=ALL-UNNAMED --add-opens=jdk.management/com.sun.management.internal=ALL-UNNAMED -Xmx8g -Xms8g -XX:+HeapDumpOnOutOfMemoryError -Dhttp.nonProxyHosts=localhost|127.*|[::1] -cp ./lib/sonar-application-25.3.0.104237.jar;C:\tools\sonarqube\lib\jdbc\postgresql\postgresql-42.7.5.jar org.sonar.server.app.WebServer c:\tools\sonarqube\temp\sq-process7939811376960153134properties
Standard Commons Logging discovery in action with spring-jcl: please remove commons-logging.jar from classpath in order to avoid potential conflicts
2025.03.16 16:28:14 INFO  app[][o.s.a.SchedulerImpl] Process[Web Server] is stopped
2025.03.16 16:28:14 INFO  app[][o.s.a.SchedulerImpl] Process[ElasticSearch] is stopped
2025.03.16 16:28:14 INFO  app[][o.s.a.SchedulerImpl] SonarQube is stopped

**Comment by areunovSennder** [comment]
@killallxxx , работает с 25.1.0

**Comment by ilya2184** [comment]
> Добрый день! SonarQube Community Build v25.3.0.104237

[ sonarqube:25.2.0.102705-community](https://hub.docker.com/layers/library/sonarqube/25.2.0.102705-community/images/sha256-1792e787f1dc2620c0eb47c5c116adc52276279a73011903598649ff98d50440) - works ok:
 - sonar-communitybsl-plugin-1.15.2.jar
 - sonarqube-community-branch-plugin-1.23.0.jar

**Comment by qathy** [comment]
Also experiencing this same issue after a SQ CE migration from 25.1.0 to 25.3.0, I have sort of came to find a workaround:
Project Settings | Background Tasks --> enable to select the branch on which an analysi ran.

But, yes, the main branch selector no longer enable to switch easily from a branch to another in main GUI views.

Hope anyway the workaround may help the community using sonarqube-community-branch-plugin-1.23.0.jar along with sonarqube-25.3.0.104237

**Comment by safetylab** [comment]
any updates for sonarqube-25.3.0.104237 ?

**Comment by chenma18** [comment]
We also had this problem with this sonarqube-25.3.0.104237 version and wanted to wait for a solution

**Comment by MCMicS** [comment]
with `25.4.0.105899` I see now an error
```
2025.04.09 10:35:51 ERROR web[][o.s.s.p.w.RootFilter] Processing of request /api/project_pull_requests/list?project=sample failed
jakarta.servlet.ServletException: Filter execution threw an exception
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:171)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
        at org.sonar.server.platform.web.UserSessionFilter.doFilter(UserSessionFilter.java:83)
        at org.sonar.server.platform.web.UserSessionFilter.doFilter(UserSessionFilter.java:70)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
        at org.sonar.server.platform.web.CspFilter.doFilter(CspFilter.java:67)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
        at org.sonar.server.platform.web.CacheControlFilter.doFilter(CacheControlFilter.java:76)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
        at org.sonar.server.platform.web.SecurityServletFilter.doHttpFilter(SecurityServletFilter.java:60)
        at org.sonar.server.platform.web.SecurityServletFilter.doFilter(SecurityServletFilter.java:47)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
        at org.sonar.server.platform.web.RedirectFilter.doFilter(RedirectFilter.java:56)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
        at org.sonar.server.platform.web.EndpointPathFilter.doFilter(EndpointPathFilter.java:47)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
        at org.sonar.server.platform.web.RequestIdFilter.doFilter(RequestIdFilter.java:66)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
        at org.sonar.server.platform.web.RootFilter.doFilter(RootFilter.java:65)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
        at org.apache.catalina.filters.SetCharacterEncodingFilter.doFilter(SetCharacterEncodingFilter.java:115)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
        at ch.qos.logback.access.tomcat.LogbackValve.invoke(LogbackValve.java:268)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
        at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
        at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NoSuchMethodError: 'boolean org.sonar.server.user.UserSession.hasEntityPermission(java.lang.String, org.sonar.db.entity.EntityDto)'
        at com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.action.ListAction.checkPermission(ListAction.java:108)
        at com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.action.ListAction.handleProjectRequest(ListAction.java:79)
        at com.github.mc1arke.sonarqube.plugin.server.pullrequest.ws.pullrequest.action.ProjectWsAction.handle(ProjectWsAction.java:61)
        at org.sonar.server.ws.WebServiceEngine.execute(WebServiceEngine.java:114)
        at org.sonar.server.platform.web.WebServiceFilter.doFilter(WebServiceFilter.java:84)
        at org.sonar.server.platform.web.MasterServletFilter$JavaxFilterAdapter.doFilter(MasterServletFilter.java:194)
        at org.sonar.server.platform.web.MasterServletFilter$GodFilterChain.doFilter(MasterServletFilter.java:165)
        at org.sonar.server.platform.web.MasterServletFilter$HttpFilterChainAdapter.doFilter(MasterServletFilter.java:208)
        at org.sonar.server.platform.web.SonarQubeIdeConnectionFilter.doFilter(SonarQubeIdeConnectionFilter.java:62)
        at org.sonar.server.platform.web.MasterServletFilter$JavaxFilterAdapter.doFilter(MasterServletFilter.java:194)
        at org.sonar.server.platform.web.MasterServletFilter$GodFilterChain.doFilter(MasterServletFilter.java:165)
        at org.sonar.server.platform.web.MasterServletFilter.doFilter(MasterServletFilter.java:126)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
        ... 47 common frames omitted
```

**Comment by ngeleedst** [comment]
Hi,
Any updates regarding this issue. We are currently stuck between versions. I already tried downgrading SQ but it's giving errors because of mismatch due DB migrations...

**Comment by arihantsingh1310** [comment]
Hello, we have been waiting for this issue to be resolved as well. Is there any update regarding the issue or any alternative available?

**Comment by WalidSlamaVDM** [comment]
Any update on this issue? I have the problem issue with sonarqube v25.5.0.107428 , can't swith to devleop branch. when I go to background tasks i can see the task on the develop branch was success

![Image](https://github.com/user-attachments/assets/cfd5a046-4347-4eb5-8e3b-40620e1a9220)

![Image](https://github.com/user-attachments/assets/496ee07d-84a9-40ac-91bb-b4faeab9650a)

**Comment by nils2504** [comment]
I do see the error still exists in v25.5.0.107428 - event with the 1.24.0 from bernalvarela...
Any update on that?

**Comment by bernalvarela** [comment]
My commit solves the bugfix-1005. This bug is not related with my pr.

**Comment by leonardfactory** [comment]
> I think this is likely to be [SonarSource/sonarqube-webapp@d5214e1](https://github.com/SonarSource/sonarqube-webapp/commit/d5214e1472063e4985ffc70eb59fda0fa30dbbe9) which has removed parts of the branch UI from the community edition front-end. Whilst plugins can contribute pages to the UI, I don't know if it will be possible to inject components back into the core framework. Interestingly it is possibly to still load issues on branches and pull requests by adding a `pullRequest={id}` or `branch={id}` parameter to the URL, so the data is still being read from the APIs properly - but the overview don't load from a quick test so I presume there have been other front-end changes to limit the data that's being retrieved or rendered for some operations.
> 
> I'll have a go at seeing what it's possible to hack together, but don't hold out to much hope of this working or being a long-term viable solution. Wish me luck!

Just to add on this, it seems major work has been done in this commit https://github.com/SonarSource/sonarqube-webapp/commit/e2d7902eb7ec8951140b0ef59240ec7d106eb8a8#diff-17992b0b9b0c6d96e6195b1ece27b642903c3226a1568bd542c8dde68b3565b3, where the `branch` feature has been moved into the private `addons` package.

I'm wondering if we could inject the `addons.branches` at runtime, starting from a JS file (like adding a plugin Page and using its startup code to register components), like these:

```tsx
const Menu = addons.branches?.Menu || (() => undefined);
const PRLink = (isPullRequest(currentBranchLike) && addons.branches?.PRLink) || (() => undefined);
```

This would still require to adapt/develop removed OSS code in a plugin, so it doesn't seem an easy task. Plus we don't know further changes made in the private repository. I'd like to know your thoughts on that and if there's some way we could help.

**Comment by lfdominguez** [comment]
Hummm I think that this is a bad move from Sonar, they don't like that there is a plugin that prevents buying their license, something as basic as multi-branch analysis they want to charge for, so I think they will gradually move more and more components to private.

**Comment by mc1arke** [comment]
Fix release in 1.24.0 of the plugin to support the 25.3 Sonarqube UI branch features

**Comment by strowi** [comment]
@mc1arke Thank you for the work! Just to confirm, besides rust-plugin breaking the upgrade and giving me a small panic, it seems to work flawless.  ;)

**Comment by simonas-opti** [comment]
--@mc1arke this issue seems to have reappeared again with 25.6-- missed a step in plugin upgrade

**Comment by adrianmusante** [comment]
> --[@mc1arke](https://github.com/mc1arke) this issue seems to have reappeared again with 25.6-- missed a step in plugin upgrade

@simonas-opti maybe you forgot to update the web module. It works in the version I built. Are you using a pre-built image or are you doing it yourself?

![Image](https://github.com/user-attachments/assets/402038af-6e6a-47be-a52e-4ca08e66cafb)

**Comment by simonas-opti** [comment]
> > --[@mc1arke](https://github.com/mc1arke) this issue seems to have reappeared again with 25.6-- missed a step in plugin upgrade
> 
> [@simonas-opti](https://github.com/simonas-opti) maybe you forgot to update the web module. It works in the version I built. Are you using a pre-built image or are you doing it yourself?
> 
> ![Image](https://github.com/user-attachments/assets/402038af-6e6a-47be-a52e-4ca08e66cafb)

@adrianmusante exactly that, all sorted now
